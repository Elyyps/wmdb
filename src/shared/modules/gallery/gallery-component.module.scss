.gallery {
  &-images {
    @media (min-width: $breakpoint-small) {
      margin: 0 -8px 0 -8px;
    }
    &-item {
      height: 160px;
      width: 213px;
      z-index: 10;
      img {
        object-fit: cover;
        width: 100%;
        height: 100%;
        border-radius: $global-border-radius;
      }
    }
  }
  &__lightbox {
    width: 100%;
    position: absolute;
    top: 12%;
    height: 60%;
    display: flex;
    justify-content: center;
    align-items: center;
    &-item {
      @media (max-width: $breakpoint-medium) {
        max-width: 70%;
      }
      @media (max-width: $breakpoint-xsmall-max) {
        width: 230px;
        height: 110px;
      }
      &-video {
        margin: 0 auto;
        width: 700px;
        height: 450px;

        @media (max-width: $breakpoint-medium) {
          max-width: 100%;
          height: 350px;
        }
        @media (max-width: $breakpoint-xsmall-max) {
          width: 230px;
          height: 110px;
        }
      }
      img {
        width: inherit;
        height: inherit;
      }
    }
    &-arrows {
      &-next {
        right: 3%;
      }
      &-prev {
        left: 3%;
      }
      div {
        top: 50%;
        position: absolute;
        cursor: pointer;
        height: 48px;
        width: 48px;
        opacity: 1;
        -webkit-transform: translateY(-50%);
        transform: translateY(-50%);
        -webkit-transition: opacity 0.3s;
        transition: opacity 0.3s;
        background-color: rgba(0, 0, 0, 0.5);
        border-radius: 50%;
        border: none;
        span {
          margin: auto;
        }
      }
    }
  }
  &__info {
    width: fit-content;
    position: relative;
    bottom: 0px;
    left: 0;
    background-color: $global-inverse-color;
    border: solid 1px #eaeaea;
    box-shadow: 0 6px 12px 0 rgba($black, 0.06);
    padding: 12px 16px;
    border-radius: $global-border-radius;

    .slider-info {
      display: flex;
      align-items: center;
      margin: 0 -8px;

      > * {
        font-size: 14px;
        font-weight: 600;
        padding: 0 8px;
        display: flex;
        align-items: center;

        span {
          margin-right: 8px;
        }
      }
      .photo {
        color: $brand-primary;
        svg {
          path {
            fill: $brand-primary;
          }
        }
      }
      .video {
        cursor: pointer;
      }
    }
  }
}
