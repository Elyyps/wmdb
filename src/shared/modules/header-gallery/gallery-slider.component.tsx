import React from "react";
import Slider, { Settings } from "react-slick";
import { BasicArrow } from "./basic-arrow.component";
import { ImageComponent } from "@app/core/image";

export interface IGallerySliderComponentProps {
  currentIndex: number;
  images: string[];
}

const GallerySliderComponent = ({ currentIndex, images }: IGallerySliderComponentProps) => {
  const ref: any = React.createRef();

  const settings: Settings = {
    dots: false,
    arrows: true,
    infinite: true,
    prevArrow: <BasicArrow prev={true} />,
    nextArrow: <BasicArrow />,
    responsive: [
      {
        breakpoint: 1199,
        settings: {
          arrows: false,
          dots: true,
          slidesToScroll: 1,
          slidesToShow: 1
        }
      }
    ],
    slidesToScroll: 1,
    slidesToShow: 1,
    speed: 500,
    initialSlide: currentIndex
  };

  return (
    <Slider ref={ref} {...settings} initialSlide={currentIndex}>
      {images.map((item: string, key: number) => (
        <ImageComponent src={item} key={key} />
      ))}
    </Slider>
  );
};

export { GallerySliderComponent };
