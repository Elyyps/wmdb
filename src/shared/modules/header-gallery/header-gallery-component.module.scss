.header-gallery {
  &__container {
    display: flex;
    margin: 0 -32px;
    border-radius: $global-border-radius;
    overflow: hidden;
  }
  &__slider_images {
    img {
      height: 496px !important;
    }
  }
  &__slider_arrows {
    position: absolute;
    top: 50%;
    height: 50px;
    width: 100%;
    div {
      position: absolute;
      cursor: pointer;
      height: 48px;
      width: 48px;
      opacity: 1;
      -webkit-transform: translateY(-50%);
      transform: translateY(-50%);
      -webkit-transition: opacity 0.3s;
      transition: opacity 0.3s;
      background-color: rgba(0, 0, 0, 0.5);
      border-radius: 50%;
      border: none;
      span {
        margin: auto;
      }
    }
    &_prev {
      left: 10px;
    }
    &_next {
      right: 10px;
    }
  }

  &__col {
    box-sizing: border-box;
    &-card {
      img {
        height: 248px;
      }
    }
    &-cards {
      img {
        height: 116px;
      }
    }
    img {
      object-fit: cover;
      width: 100%;
    }
    &.large {
      width: 760px;
      position: relative;

      @media (max-width: $breakpoint-medium-max) {
        width: 100%;
      }
    }

    &.small {
      flex: 1;
    }
    .uk-grid-small {
      margin-left: 0;
      height: 496px;
      > * {
        padding-left: 8px;
      }

      > *:not(:first-child) {
        padding-top: 8px;
      }
    }
  }
  &__info {
    position: absolute;
    bottom: 24px;
    left: 24px;
    background-color: $global-inverse-color;
    box-shadow: 0 6px 12px 0 rgba($black, 0.06);
    padding: 12px 16px;
    border-radius: $global-border-radius;

    .slider-info {
      display: flex;
      align-items: center;
      margin: 0 -8px;
      > * {
        font-size: 14px;
        font-weight: 600;
        padding: 0 8px;
        display: flex;
        align-items: center;

        span {
          margin-right: 8px;
        }
      }
      .photo {
        color: $brand-primary;
        svg {
          path {
            fill: $brand-primary;
          }
        }
      }
    }
  }
}
