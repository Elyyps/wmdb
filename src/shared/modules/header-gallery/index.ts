export * from "./header-gallery.component";
