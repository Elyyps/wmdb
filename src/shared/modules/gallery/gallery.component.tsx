import * as React from "react";
import styles from "./gallery-component.module.scss";
import { ImageComponent } from "@app/core/image";
import { ILink } from "@app/api/core/link";
import { Breadcrumb } from "@app/core/breadcrumb";
import { IconComponent } from "@app/core/icon";
import IconImage from "@assets/icons/images.svg";
import IconPlay from "@assets/icons/play.svg";
import { getArrow } from "@app/constants/icons";
import ReactPlayer from "react-player";
import { IHeaderGallery } from "@app/api/modules/header-gallery/header-gallery";
import { LightBoxComponent } from "@app/core/lightbox";
import ReactAliceCarousel from "react-alice-carousel";
import "react-alice-carousel/lib/alice-carousel.css";





export interface IGalleryComponentProps {
  headerGallery: IHeaderGallery[];
}
const BreadcrumbsData: ILink[] = [
  {
    title: "Home",
    url: "/"
  },
  {
    title: "Citygames in Amsterdam",
    url: "/overzicht"
  },
  {
    title: "Wie is de mol?",
    url: ""
  }
];
const GalleryComponent = ({ headerGallery }: IGalleryComponentProps) => {
  const [lightBoxOpen, setLightBoxOpen] = React.useState(false);
  const [currentSlide, setCurrentSlide] = React.useState<number>(0);
  const contentHeight = 50;

  const handleClick = (newSlide: number) => {
    setCurrentSlide(newSlide);
    setLightBoxOpen(!lightBoxOpen);
  };
  const findFirstElement = (type?: "video" | "image") => {
    const firstElement = headerGallery.find(item => item.type === type);
    firstElement && setCurrentSlide(firstElement.id);
  };
  const findSlide = (id: number) => {
    const slide = headerGallery.find(item => item.id === id);

    return slide ? slide : headerGallery[0];
  };

  const onButtonClick = (newImage: number) => {
    setCurrentSlide(newImage);
  };
  const responsive = {
    1200: {
      items: 5
    },
    639: {
      items: 3
    },
    0: {
      items: 1
    }
  };
  React.useEffect(() => {
    currentSlide >= headerGallery.length && setCurrentSlide(0);
    currentSlide < 0 && setCurrentSlide(headerGallery.length - 1);
  }, [currentSlide]);

  return (
    <div className={styles["gallery"]}>
      <div className="uk-container">
        <Breadcrumb breadcrumbs={BreadcrumbsData} />
        <div className={styles["gallery-images"]}>
          <ReactAliceCarousel
            onSlideChanged={e => currentSlide && onButtonClick(e.slide)}
            dotsDisabled
            buttonsDisabled
            responsive={responsive}
            startIndex={currentSlide}
            items={headerGallery.map((slide, key) => (
              <div
                role="button"
                key={key}
                className={styles["gallery-images-item"]}
                onClick={() => {
                  handleClick(slide.id);
                }}
              >
                {slide.type === "image" ? (
                  <ImageComponent src={slide.url} />
                ) : (
                  slide.type === "video" ? (
                    <ReactPlayer
                      style={{ pointerEvents: "none" }}
                      url={slide.url}
                      height={"100%"}
                      width={"100%"}
                      controls
                      light
                    />
                    
                  )
                  :""
                )}
              </div>
            ))}
          />
          <div className={` ${styles["gallery__slider_arrows"]} uk-visible@s`}>
            <div
              className={styles["gallery__slider_arrows_prev"]}
              onClick={() => setCurrentSlide(currentSlide - 1)}
              role="button"
            >
              <IconComponent icon={getArrow(true)} size={"8px"} fillColor={"white"} />
            </div>
            <div
              className={styles["gallery__slider_arrows_next"]}
              onClick={() => setCurrentSlide(currentSlide + 1)}
              role="button"
            >
              <IconComponent icon={getArrow(false)} size={"8px"} fillColor={"white"} />
            </div>
          </div>
        </div>
        <div className={`${styles["gallery__info"]} uk-visible@s`}>
          <div className={styles["slider-info"]}>
            <div
              role="button"
              className={styles["photo"]}
              style={{ cursor: "pointer" }}
              onClick={() => findFirstElement("image")}
            >
              <IconComponent icon={IconImage} size={"15px"} />
              Fotoâ€™s ({headerGallery.length})
            </div>
            <div role="button" style={{ cursor: "pointer" }} onClick={() => findFirstElement("video")}>
              <IconComponent icon={IconPlay} size={"12px"} />
              Video
            </div>
          </div>
        </div>
      </div>

      {lightBoxOpen && currentSlide !== undefined && (
        <LightBoxComponent
          heightLightBoxContent={contentHeight}
          isLightBoxOpen={lightBoxOpen}
          setLightBoxOpen={() => setLightBoxOpen(!lightBoxOpen)}
        >
          <div className={styles["gallery__lightbox"]}>
            <div className={styles["gallery__lightbox-item"]}>
              {findSlide(currentSlide).type === "image" ? (
                <ImageComponent src={findSlide(currentSlide).url} />
              ) : (
                findSlide(currentSlide).type === "video" && (
                  <div className={styles["gallery__lightbox-item-video"]}>
                    <ReactPlayer url={findSlide(currentSlide).url} width={"100%"} height={"100%"} />
                  </div>
                )
              )}
            </div>

            <div className={styles["gallery__lightbox-arrows"]}>
              <div
                className={styles["gallery__lightbox-arrows-prev"]}
                onClick={() => setCurrentSlide(currentSlide - 1)}
                role="button"
              >
                <IconComponent icon={getArrow(true)} size={"8px"} fillColor={"white"} />
              </div>

              <div
                className={styles["gallery__lightbox-arrows-next"]}
                onClick={() => setCurrentSlide(currentSlide + 1)}
                role="button"
              >
                <IconComponent icon={getArrow(false)} size={"8px"} fillColor={"white"} />
              </div>
            </div>
          </div>
        </LightBoxComponent>
      )}
    </div>
  );
};

export { GalleryComponent };
