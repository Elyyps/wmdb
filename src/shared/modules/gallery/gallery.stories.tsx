import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { GalleryComponent } from "./gallery.component";
import { generateGalleryData } from "@app/api/modules/header-gallery/end-point";

storiesOf("Gallery", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => <GalleryComponent headerGallery={generateGalleryData()} />);
