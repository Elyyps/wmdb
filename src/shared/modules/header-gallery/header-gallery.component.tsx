import * as React from "react";
import styles from "./header-gallery-component.module.scss";
import { IconComponent } from "@app/core/icon";
import { ImageComponent } from "@app/core/image";
import IconImage from "@assets/icons/images.svg";
import IconPlay from "@assets/icons/play.svg";
import AliceCarousel from "react-alice-carousel";
import "react-alice-carousel/lib/alice-carousel.css";
import { getArrow } from "@app/constants/icons";
import { ILink } from "@app/api/core/link";
import { Breadcrumb } from "@app/core/breadcrumb";
import { IHeaderGallery } from "@app/api/modules/header-gallery/header-gallery";
import ReactPlayer from "react-player";

export interface IHeaderGalleryComponentProps {
  headerGallery: IHeaderGallery[];
}

const BreadcrumbsData: ILink[] = [
  {
    title: "Home",
    url: "/"
  },
  {
    title: "Citygames in Amsterdam",
    url: "/overzicht"
  },
  {
    title: "Wie is de mol?",
    url: ""
  }
];

const HeaderGalleryComponent = ({ headerGallery }: IHeaderGalleryComponentProps) => {
  const [currentImage, setCurrentImage] = React.useState(0);
  const [videoIsPlaying, setVideoIsPlaying] = React.useState(false);
  const [buttonActive, setbuttonActive] = React.useState<undefined|"image"|"video">(undefined);
  const smallImagesStart = 2;

  const responsive = {
    desktop: {
      breakpoint: { max: 3000, min: 1024 },
      items: 1
    }
  };

  const onVideoClick = () => {
    setVideoIsPlaying(true);
  };
  const onButtonClick = (newImage: number) => {
    setCurrentImage(newImage);
    setVideoIsPlaying(false);
  };

  const findFirstElement = (type: "video" | "image") => {
    const firstElement = headerGallery.find(item => item.type === type);
    firstElement && setCurrentImage(firstElement.id);
    setbuttonActive(type);
  };

  React.useEffect(() => {
    currentImage >= headerGallery.length && onButtonClick(0);
    currentImage < 0 && onButtonClick(headerGallery.length - 1);
  }, [currentImage]);

  return (
    <div className={styles["header-gallery"]}>
      <div className="uk-container">
        <Breadcrumb breadcrumbs={BreadcrumbsData} />
        <div className={styles["header-gallery__container"]}>
          <div className={`${styles["header-gallery__col"]} ${styles["large"]}`}>
            <div className={styles["header-gallery__slider_images"]}>
              <AliceCarousel
                onSlideChanged={e => onButtonClick(e.slide)}
                dotsDisabled
                startIndex={currentImage}
                responsive={responsive}
                infinite={true}
                buttonsDisabled
                items={headerGallery.map((item: IHeaderGallery, key: number) =>
                  item.type === "image" ? (
                    <ImageComponent src={item.url} key={key} />
                  ) : (
                    <ReactPlayer
                      url={item.url}
                      width={"100%"}
                      height={"496px"}
                      controls
                      onPlay={onVideoClick}
                      playing={currentImage === item.id && videoIsPlaying && true}
                    />
                    
                  )
                )}
              />
              <div className={styles["header-gallery__slider_arrows"]}>
                <div
                  className={styles["header-gallery__slider_arrows_prev"]}
                  onClick={() => onButtonClick(currentImage - 1)}
                  role="button"
                >
                  <IconComponent icon={getArrow(true)} size={"8px"} fillColor={"white"} />
                </div>
                <div
                  className={styles["header-gallery__slider_arrows_next"]}
                  onClick={() => onButtonClick(currentImage + 1)}
                  role="button"
                >
                  <IconComponent icon={getArrow(false)} size={"8px"} fillColor={"white"} />
                </div>
              </div>
            </div>
            {!videoIsPlaying && (
              <div className={`${styles["header-gallery__info"]} ${"uk-visible@l"}`}>
                <div className={styles["slider-info"]}>
                  <div
                    role="button"
                    className={styles["photo"]}
                    style={{ cursor: "pointer",color:buttonActive==="image" ? "#34aadf" :"black"  }}
                    onClick={() => findFirstElement("image")}
                  >
                    <IconComponent icon={IconImage} size={"15px"} fillColor={buttonActive==="image" ? "#34aadf" :"black" }/>
                    Fotoâ€™s ({currentImage + 1}/{headerGallery.length})
                  </div>
                  <div role="button" style={{ cursor: "pointer",color:buttonActive==="video" ? "#34aadf" :"black"  }} onClick={() => findFirstElement("video")}>
                    <IconComponent icon={IconPlay} size={"12px"} fillColor={buttonActive==="video" ? "#34aadf" :"black" }/>
                    Video
                  </div>
                </div>
              </div>
            )}
          </div>
          <div className={` ${styles["header-gallery__col"]} ${styles["small"]} ${"uk-visible@l"} `}>
            <div className={` uk-grid ${styles["uk-grid-small"]} data-uk-margin `}>
              {
                <>
                  <div
                    role="button"
                    className={` ${styles["header-gallery__col-card"]} uk-width-1-1`}
                    style={{ cursor: "pointer" }}
                    onClick={() => onButtonClick(1)}
                  >
                    {headerGallery[1].type === "image" ? (
                      <ImageComponent src={headerGallery[1].url} />
                    ) : (
                      <ReactPlayer
                        url={headerGallery[1].url}
                        width={"100%"}
                        height={"248px"}
                        controls
                        light
                        style={{ pointerEvents: "none" }}
                      />
                      
                    )}
                  </div>
                  {headerGallery
                    .slice(smallImagesStart, headerGallery.length)
                    .map((value: IHeaderGallery, key: number) => (
                      <div
                        className={` ${styles["header-gallery__col-cards"]} uk-width-1-2@s`}
                        key={key}
                        style={{ cursor: "pointer" }}
                        onClick={() => onButtonClick(key + smallImagesStart)}
                        role="button"
                      >
                        {value.type === "image" ? (
                          <ImageComponent src={value.url} />
                        ) : (
                          value.type === "video" && (
                            <ReactPlayer
                              url={value.url}
                              width={"100%"}
                              height={"116px"}
                              controls
                              light
                              style={{ pointerEvents: "none" }}
                            />
                           
                          )
                        )}
                      </div>
                    ))}
                </>
              }
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export { HeaderGalleryComponent };
