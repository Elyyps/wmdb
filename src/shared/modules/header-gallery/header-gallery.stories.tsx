import * as React from "react";
import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { HeaderGalleryComponent } from "./header-gallery.component";

import { generateGalleryData } from "@app/api/modules/header-gallery/end-point";

storiesOf("HeaderGallery", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => <HeaderGalleryComponent headerGallery={generateGalleryData()} />);
