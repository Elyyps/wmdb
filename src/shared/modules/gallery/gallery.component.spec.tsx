import React from "react";
import { shallow } from "enzyme";
import { GalleryComponent } from "./gallery.component";
import { generateGalleryData } from "@app/api/modules/header-gallery/end-point";

describe("[Gallery]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<GalleryComponent galleryModule={generateGalleryData()} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
