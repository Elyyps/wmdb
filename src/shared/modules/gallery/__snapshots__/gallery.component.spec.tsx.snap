// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Gallery] should render component without crashing 1`] = `
<div
  className="gallery"
>
  <div
    className="uk-container"
  >
    <Breadcrumb
      breadcrumbs={
        Array [
          Object {
            "title": "Home",
            "url": "/",
          },
          Object {
            "title": "Citygames in Amsterdam",
            "url": "/overzicht",
          },
          Object {
            "title": "Wie is de mol?",
            "url": "",
          },
        ]
      }
    />
    <div
      className="gallery-images"
    >
      <AliceCarousel
        autoHeight={false}
        autoPlay={false}
        autoPlayDirection="ltr"
        autoPlayInterval={250}
        buttonsDisabled={true}
        disableAutoPlayOnAction={false}
        dotsDisabled={true}
        duration={250}
        fadeOutAnimation={false}
        infinite={true}
        items={
          Array [
            <div
              className="gallery-images-item"
              onClick={[Function]}
              role="button"
            >
              <ImageComponent
                src="img-1.jpg"
              />
            </div>,
            <div
              className="gallery-images-item"
              onClick={[Function]}
              role="button"
            >
              <ImageComponent
                src="img-2.png"
              />
            </div>,
            <div
              className="gallery-images-item"
              onClick={[Function]}
              role="button"
            >
              <ImageComponent
                src="img-3.png"
              />
            </div>,
            <div
              className="gallery-images-item"
              onClick={[Function]}
              role="button"
            >
              <ImageComponent
                src="img-4.png"
              />
            </div>,
            <div
              className="gallery-images-item"
              onClick={[Function]}
              role="button"
            >
              <ImageComponent
                src="img-5.png"
              />
            </div>,
            <div
              className="gallery-images-item"
              onClick={[Function]}
              role="button"
            >
              <ImageComponent
                src="img-6.png"
              />
            </div>,
            <div
              className="gallery-images-item"
              onClick={[Function]}
              role="button"
            >
              <ReactPlayer
                config={
                  Object {
                    "dailymotion": Object {
                      "params": Object {
                        "api": 1,
                        "endscreen-enable": false,
                      },
                      "preload": false,
                    },
                    "facebook": Object {
                      "appId": "1309697205772819",
                      "version": "v3.3",
                    },
                    "file": Object {
                      "attributes": Object {},
                      "dashVersion": "2.9.2",
                      "forceAudio": false,
                      "forceDASH": false,
                      "forceHLS": false,
                      "forceVideo": false,
                      "hlsOptions": Object {},
                      "hlsVersion": "0.10.1",
                      "tracks": Array [],
                    },
                    "mixcloud": Object {
                      "options": Object {
                        "hide_cover": 1,
                      },
                    },
                    "soundcloud": Object {
                      "options": Object {
                        "buying": false,
                        "download": false,
                        "liking": false,
                        "sharing": false,
                        "show_comments": false,
                        "show_playcount": false,
                        "visual": true,
                      },
                    },
                    "twitch": Object {
                      "options": Object {},
                    },
                    "vimeo": Object {
                      "playerOptions": Object {
                        "autopause": false,
                        "byline": false,
                        "portrait": false,
                        "title": false,
                      },
                      "preload": false,
                    },
                    "wistia": Object {
                      "options": Object {},
                    },
                    "youtube": Object {
                      "embedOptions": Object {},
                      "playerVars": Object {
                        "iv_load_policy": 3,
                        "modestbranding": 1,
                        "playsinline": 1,
                        "rel": 0,
                        "showinfo": 0,
                      },
                      "preload": false,
                    },
                  }
                }
                controls={true}
                height="100%"
                light={true}
                loop={false}
                muted={false}
                onBuffer={[Function]}
                onBufferEnd={[Function]}
                onDisablePIP={[Function]}
                onDuration={[Function]}
                onEnablePIP={[Function]}
                onEnded={[Function]}
                onError={[Function]}
                onPause={[Function]}
                onPlay={[Function]}
                onProgress={[Function]}
                onReady={[Function]}
                onSeek={[Function]}
                onStart={[Function]}
                pip={false}
                playbackRate={1}
                playing={false}
                playsinline={false}
                progressInterval={1000}
                style={
                  Object {
                    "pointerEvents": "none",
                  }
                }
                url="https://www.youtube.com/watch?v=7gphiFVVtUI"
                volume={null}
                width="100%"
                wrapper="div"
              />
            </div>,
            <div
              className="gallery-images-item"
              onClick={[Function]}
              role="button"
            >
              <ReactPlayer
                config={
                  Object {
                    "dailymotion": Object {
                      "params": Object {
                        "api": 1,
                        "endscreen-enable": false,
                      },
                      "preload": false,
                    },
                    "facebook": Object {
                      "appId": "1309697205772819",
                      "version": "v3.3",
                    },
                    "file": Object {
                      "attributes": Object {},
                      "dashVersion": "2.9.2",
                      "forceAudio": false,
                      "forceDASH": false,
                      "forceHLS": false,
                      "forceVideo": false,
                      "hlsOptions": Object {},
                      "hlsVersion": "0.10.1",
                      "tracks": Array [],
                    },
                    "mixcloud": Object {
                      "options": Object {
                        "hide_cover": 1,
                      },
                    },
                    "soundcloud": Object {
                      "options": Object {
                        "buying": false,
                        "download": false,
                        "liking": false,
                        "sharing": false,
                        "show_comments": false,
                        "show_playcount": false,
                        "visual": true,
                      },
                    },
                    "twitch": Object {
                      "options": Object {},
                    },
                    "vimeo": Object {
                      "playerOptions": Object {
                        "autopause": false,
                        "byline": false,
                        "portrait": false,
                        "title": false,
                      },
                      "preload": false,
                    },
                    "wistia": Object {
                      "options": Object {},
                    },
                    "youtube": Object {
                      "embedOptions": Object {},
                      "playerVars": Object {
                        "iv_load_policy": 3,
                        "modestbranding": 1,
                        "playsinline": 1,
                        "rel": 0,
                        "showinfo": 0,
                      },
                      "preload": false,
                    },
                  }
                }
                controls={true}
                height="100%"
                light={true}
                loop={false}
                muted={false}
                onBuffer={[Function]}
                onBufferEnd={[Function]}
                onDisablePIP={[Function]}
                onDuration={[Function]}
                onEnablePIP={[Function]}
                onEnded={[Function]}
                onError={[Function]}
                onPause={[Function]}
                onPlay={[Function]}
                onProgress={[Function]}
                onReady={[Function]}
                onSeek={[Function]}
                onStart={[Function]}
                pip={false}
                playbackRate={1}
                playing={false}
                playsinline={false}
                progressInterval={1000}
                style={
                  Object {
                    "pointerEvents": "none",
                  }
                }
                url="https://www.youtube.com/watch?v=2oyhlad64-s"
                volume={null}
                width="100%"
                wrapper="div"
              />
            </div>,
          ]
        }
        keysControlDisabled={false}
        mouseDragEnabled={false}
        onSlideChanged={[Function]}
        playButtonEnabled={false}
        preventEventOnTouchMove={false}
        responsive={
          Object {
            "0": Object {
              "items": 1,
            },
            "1200": Object {
              "items": 5,
            },
            "639": Object {
              "items": 3,
            },
          }
        }
        showSlideInfo={false}
        slideToIndex={0}
        stagePadding={Object {}}
        startIndex={0}
        stopAutoPlayOnHover={true}
        swipeDisabled={false}
      />
      <div
        className=" gallery__slider_arrows uk-visible@s"
      >
        <div
          className="gallery__slider_arrows_prev"
          onClick={[Function]}
          role="button"
        >
          <IconComponent
            fillColor="white"
            icon={
              Object {
                "ReactComponent": [Function],
                "__esModule": true,
                "default": "chevron-left.svg",
              }
            }
            size="8px"
          />
        </div>
        <div
          className="gallery__slider_arrows_next"
          onClick={[Function]}
          role="button"
        >
          <IconComponent
            fillColor="white"
            icon={
              Object {
                "ReactComponent": [Function],
                "__esModule": true,
                "default": "chevron-right.svg",
              }
            }
            size="8px"
          />
        </div>
      </div>
    </div>
    <div
      className="gallery__info uk-visible@s"
    >
      <div
        className="slider-info"
      >
        <div
          className="photo"
          onClick={[Function]}
          role="button"
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <IconComponent
            icon="images.svg"
            size="15px"
          />
          Foto’s (
          8
          )
        </div>
        <div
          onClick={[Function]}
          role="button"
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <IconComponent
            icon="play.svg"
            size="12px"
          />
          Video
        </div>
      </div>
    </div>
  </div>
</div>
`;
